compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp

```diff
@@ -259,7 +271,13 @@ static FailureOr<unsigned> fuseMultiUseProducers(Operation *funcOp,
             continue;
           }
 
-          // 7. All uses from `producer` -> `consumer` need to be fusable.
+          // 7. Skip dequantization-like `producer` ops as we would rather fuse
+          // by cloning the producer instead of multi-use fusion.
+          if (isDequantizationLikeOp(producer)) {
+            return;
+          }
+
+          // 8. All uses from `producer` -> `consumer` need to be fusable.
```

Comment:
Why is it better to clone the producer dequant into a dispatch region rather than elementwise fuse it here?

Reply:
multi-use producer fusion creates ops like
```
linalg.generic
  %0 = ...
  %1 = arith.extf %0
  return %0, %1
```
basically it returns both values that are needed by consumers. The assumption is that we'd rather just do the `arith.extf` for all consumers because it's going to be cheaper than having to allocate, write out, and load the full extended data. I'm sure we'll find cases where this heuristic gets it wrong, but this seems like a reasonable enough rule of thumb to me.

Reply:
Ah, thanks for explaining that. I didn't realize what exactly the multi-use producer fusion did. This makes sense now.